### PR TITLE
fix(core): T75 — resolveDkgHome fresh-install fallback mirrors resolveDkgConfigHome's monorepo heuristic

### PR DESCRIPTION
**Pull request by Jurij89**
## Summary

- Codex flagged on PR #264 (after #337 merged) that `resolveDkgHome` and `resolveDkgConfigHome` disagree on fresh monorepo installs. The runtime resolver returned `~/.dkg` unconditionally on no-signal fallback, while setup wrote the daemon to `~/.dkg-dev` for the same install. Adapter would cache the wrong dir for the gateway's plugin lifetime.
- Step (5) of `resolveDkgHome` now mirrors `resolveDkgConfigHome`'s monorepo heuristic: in a monorepo without a globally-installed `~/.dkg/config.json`, prefer `~/.dkg-dev`. Otherwise `~/.dkg`. Both helpers now agree on every fresh-install case.
- Added `ResolveDkgHomeOptions` interface with optional `isDkgMonorepo` test hook (mirrors `ResolveDkgConfigHomeOptions`).

## Related

- Surfaced by Codex review on PR #264 at 2026-04-30 08:59 (after #337 merge)
- Related to #264

## Files changed

| File | What |
|------|------|
| `packages/core/src/dkg-home.ts` | New step (5) monorepo heuristic mirroring `resolveDkgConfigHome`. Updated docstring. New `ResolveDkgHomeOptions` interface with optional `isDkgMonorepo` test hook. |
| `packages/core/test/dkg-home.test.ts` | Updated tests #9 / #9b to pass explicit `isDkgMonorepo: false` (npm-install case). Added 2 new T75 tests covering monorepo + no-global-config and monorepo + global-config branches. |

## Concrete failure trace this fixes

1. Fresh monorepo install. `dkg openclaw setup` runs and writes the daemon's config to `~/.dkg-dev` (via `resolveDkgConfigHome`'s monorepo branch).
2. Gateway loads BEFORE the first daemon start. No `daemon.pid` / `api.port` files anywhere.
3. `resolveDkgHome` step (5) (no signals) returned `~/.dkg` unconditionally.
4. Adapter cached `this.dkgHome = ~/.dkg` for plugin lifetime.
5. User starts the daemon → it binds in `~/.dkg-dev` (per setup's config).
6. Adapter reads `auth.token` and `agent-keystore.json` from `~/.dkg` (wrong dir) for the rest of the gateway's lifetime.

## Test plan

- [x] `pnpm -F @origintrail-official/dkg-core build` — clean.
- [x] `pnpm -F @origintrail-official/dkg-core test` — **541 passing** (537 baseline + 4 net: 2 new T75 tests, 2 existing tests with adjusted assertions).
- [x] `pnpm -F @origintrail-official/dkg-adapter-openclaw build` — clean.
- [x] `pnpm -F @origintrail-official/dkg-adapter-openclaw test` — **677 passing** + 1 known K-9 RED carryover (unchanged).
- [x] QA reviewed with 9-probe edge case matrix; PASS.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/core/src/dkg-home.ts [hunk 1]
```diff
@@ -232,7 +251,21 @@ export function resolveDkgHome(opts?: { daemonUrl?: string }): string {
...
+  //     resolver: in a monorepo checkout where `~/.dkg/config.json` does
+  //     not exist, prefer `~/.dkg-dev` (where setup will write the daemon's
+  //     config). Without this, a cold-start gateway in a fresh monorepo
+  //     install would cache `~/.dkg` while setup wrote the daemon to
+  //     `~/.dkg-dev`, and the adapter would read auth.token /
+  //     agent-keystore.json from the wrong dir for its plugin lifetime
+  //     (Codex T75 — surfaced after PR #337 introduced
+  //     `resolveDkgConfigHome` without backporting the monorepo knowledge
+  //     here).
+  const isMonorepo = opts?.isDkgMonorepo ?? findDkgMonorepoRoot() !== null;
```
[the remark this answers is not in the document]
Comment (Jurij89) [reply packages/core/src/dkg-home.ts]: Valid — fixed in 552c5113.

Replaced the inline `isMonorepo + ~/.dkg/config.json` body with a delegated call:

```ts
return resolveDkgConfigHome({ isDkgMonorepo: opts?.isDkgMonorepo });
```

`resolveDkgConfigHome` is now the single source of truth for the fresh-install monorepo / global-config rules. Any future change to those rules updates both call paths automatically — the drift surface is closed by construction.

The redundant `process.env.DKG_HOME` re-check inside `resolveDkgConfigHome` is a harmless no-op here: `resolveDkgHome` already short-circuits on env at the top, so env is guaranteed undefined when step (5) runs. The re-check returns undefined → proceeds to monorepo logic. No behavior change, no new attack surface.

Other `resolveDkgConfigHome` opts (`homeDir` / `configExists` / `startDir`) intentionally not forwarded — they're test/embedding overrides with no current consumer on the runtime path. If a future caller needs them, they're a backward-compatible addition to `ResolveDkgHomeOptions`.

QA reviewed the equivalence on all 4 cells of `(monorepo, configExists)` plus auto-detect and env-override paths — both helpers return identical values for every case. 541 core tests still pass.
